### PR TITLE
Better support for 'deep' recursive acls

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

File: lib/puppet/type/acl.rb [hunk 1]
```diff
@@ -210,11 +211,27 @@ def insync?(is)
...
+      end
+    }).to_a.flatten
+  end
+
   def newchild(path)
     full_path = ::File.join(self[:path], path)
     options = @original_parameters.merge(:name => full_path).reject { |param, value| value.nil? }
     unless File.directory?(options[:name]) then
-      options[:permission].reject! { |acl| acl.split(':', -1).length == 4 } if options.include?(:permission)
+      option[:permission] = self.class.pick_default_perms(option[:permission]) if options.include?(:permission)
```
Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: Shouldn't this be `options`?

Comment (roidelapluie) [inline lib/puppet/type/acl.rb]: Indeed forgot to push

**Comment by roidelapluie** [comment]
that is why I like travis but dislike rèuby 1.8

**Comment by dobbymoodge** [comment]
@roidelapluie lol, I hear ya

**Pull request merged**